sdk/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java

```diff
@@ -505,13 +505,7 @@ int getTotalRecordedLinks() {
     if (events.isEmpty()) {
       return Collections.emptyList();
     }
-    List<Event> result = new ArrayList<>(events.size());
-    for (io.opentelemetry.sdk.trace.TimedEvent sourceEvent : events) {
-      result.add(
-          Event.create(
-              sourceEvent.getEpochNanos(), sourceEvent.getName(), sourceEvent.getAttributes()));
-    }
-    return Collections.unmodifiableList(result);
+    return Collections.unmodifiableList(new ArrayList<Event>(events));
```

Comment:
Instances of `RawTimedEventWithEvent` should be transformed into `RawTimedEvent` to avoid holding references to memory from the `api.Event` that we store.

Reply:
ah, yes, you are correct. I'll write a unit test that verifies this, as well, since we don't have one at the moment.

Reply:
done. will re-run the benchmarks